server/src/main/java/org/opensearch/index/IndexSettings.java

```diff
@@ -1439,9 +1516,43 @@ public long getGcDeletesInMillis() {
 
     /**
      * Returns the merge policy that should be used for this index.
-     */
-    public MergePolicy getMergePolicy() {
-        return mergePolicyConfig.getMergePolicy();
+     * @param isTimeIndex true if index contains @timestamp field
+     */
+    public MergePolicy getMergePolicy(boolean isTimeIndex) {
+        String indexScopedPolicy = scopedSettings.get(INDEX_MERGE_POLICY);
+        MergePolicyProvider mergePolicyProvider = null;
+        IndexMergePolicy indexMergePolicy = IndexMergePolicy.fromString(indexScopedPolicy);
+        switch (Objects.requireNonNull(indexMergePolicy)) {
```

Comment:
The `indexMergePolicy` cannot be `null`:

```suggestion
        switch (indexMergePolicy) {
```